crates/subspace-networking/src/node_runner.rs

```diff
@@ -722,6 +722,22 @@ where
             });
 
             if full_kademlia_support {
+                let old_addresses = kademlia
```

Comment:
It seems easier to clean all the addresses and add just new ones.

Reply:
Yes, but then we'll lose information about how long peer was in k-bucket, that is the only reason I decided to do the diff instead.